File: ruby/lib/ik_classifications/null_classification.rb
# A simple test to make sure the graph is truly a Graph object and not nil.
class NullClassification < IKClassification::Base
  def classify(graph)
    graph ? CANNOT_DETERMINE_IK : IS_NOT_IK
  end
  
  def description
    "If the graph is nil, then it is NOT intrinsically knotted.  This " +
    "test is just to rule out any invalid graphs."
  end
end

File: java/src/ik/NullClassification.java
package ik;

/** 
 * A simple test to make sure the graph is truly a Graph object and not nil.
 */
public class NullClassification implements IKClassification {
  public String classify(Graph graph) {    
    return (graph == null) ? IS_NOT_IK : CANNOT_DETERMINE_IK;
  }       
  
  public String getName() {
    return "NullClassification";  
  } 
  
  public String getDescription() {
    return "If the graph is null, then it is NOT intrinsically knotted.  " +
           "This test is just to rule out any invalid graphs.";
  }
}